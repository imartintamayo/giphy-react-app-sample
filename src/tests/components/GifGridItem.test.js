import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('test <GifGridItem /> component', () => {

    const title = 'some title';
    const url = 'http://localhost/some_url.png';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('should render GifGridItem', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('should display <p>title</p>', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('<img> should have src=url and alt=title', () => {
        
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('should have animate__fadeInTopLeft class efect', () => {
        
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__fadeInTopLeft')).toBe(true);

    });
    

});