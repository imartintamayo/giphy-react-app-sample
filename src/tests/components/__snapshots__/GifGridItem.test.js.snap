// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGridItem /> component should render GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeInTopLeft"
>
  <img
    alt="some title"
    src="http://localhost/some_url.png"
  />
  <p>
    some title
  </p>
</div>
`;
